runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -857,7 +892,7 @@ ClientSessionData::getClassRecord(J9Class *clazz, JITServer::ServerStream *strea
          TR_ASSERT(it != getROMClassMap().end(), "Class %p must be already cached", clazz);
          it->second._classChainOffsetIdentifyingLoader = offset;
          it->second._classNameIdentifyingLoader = name;
-         record = getClassRecord(it->second);
+         record = getClassRecord(it->second, missingLoaderInfo);
```

Comment:
Can add a `TR_ASSERT` that `missingLoaderInfo` is false because we've just received it from the client.